main.cpp

```diff
@@ -11,28 +11,70 @@ int main(int argc, char *argv[])
     Unit *u1;
     Unit *u2;
 
-    try {
-		  if (std::string(argv[1]) == "player.json") {
-			  u1 = new Player(Unit::parseUnit(fName1));
-			  u2 = new Unit(Unit::parseUnit(fName2));
-		  }
-		  else if (std::string(argv[2]) == "player.json") {
-			  u1 = new Unit(Unit::parseUnit(fName1));
-			  u2 = new Player(Unit::parseUnit(fName2));
-		  }
-		  else {
-			  u1 = new Unit(Unit::parseUnit(fName1));
-			  u2 = new Unit(Unit::parseUnit(fName2));
-		  }
-	  }
+    try
+    {
+        if (std::string(argv[1]) == "player.json")
+        {
+            u1 = new Player(Unit::ParseUnit(fName1));
+            u2 = new Unit(Unit::ParseUnit(fName2));
+        }
+        else if (std::string(argv[2]) == "player.json")
+        {
+            u1 = new Unit(Unit::ParseUnit(fName1));
+            u2 = new Player(Unit::ParseUnit(fName2));
+        }
+        else
+        {
+            u1 = new Unit(Unit::ParseUnit(fName1));
+            u2 = new Unit(Unit::ParseUnit(fName2));
+        }
+    }
     catch (const std::string s)
     {
         std::cout << "File \"" << s << "\" not found!"
                   << std::endl;
         return 404;
     }
 
+    Unit *slowerUnit;
+    Unit *fasterUnit;
+
+    if (u1->GetAtkCoolDown() < u2->GetAtkCoolDown())
+    {
+        fasterUnit = u1;
+        slowerUnit = u2;
+    }
+    else
+    {
+        fasterUnit = u2;
+        slowerUnit = u1;
+    }
```

Comment:
Ez rendben van, de akkor utána már ne legyen használva az u1, u2, mert így átláthatatlan. Bár amúgy is kerüljön vissza a harc vezérlése a Unit osztályba.